hoomd/mpcd/update.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2009-2024 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+r"""MPCD reverse perturbation to establish nonequilibrium shear flow.
+
+These are MPCD updaters.
+
+.. invisible-code-block: python
+
+    simulation = hoomd.util.make_example_simulation(mpcd_types=["A"])
+    simulation.operations.integrator = hoomd.mpcd.Integrator(dt=0.1)
+
+"""
+
+from . import _mpcd
+from hoomd.operation import Updater
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.logging import log
+from hoomd.data.typeconverter import OnlyTypes, positive_real
+
+import math
+
+
+class ReverseNonequilibriumShearFlow(Updater):
+    r"""MPCD reverse nonequilibrium shear flow.
+
+    Args:
+        trigger (hoomd.trigger.trigger_like): Trigger to swap momentum.
+
+        num_swaps (int): Maximum number of particle pairs to swap.
+
+        slab_width (float): Thickness of momentum-exchange slabs.
+
+        target_momentum (float): Target momentum for swapped particles.
+
+    There are two exchange slabs separated by a distance of :math:`L_y/2`. The
+    edges of these slabs are located at (:math:`-L_y/2`, :math:`-L_y/2` +
+    `slab_width`) and (:math:`0.0`, `slab_width`) along the *y*-direction
+    (gradient direction) of the simulation box. On each `trigger`, particles are
+    sorted into the exchange slabs based on their positions in the box. In each
+    slab, particles whose *x*-component momenta are near `target_momentum` are
+    selected for a pairwise momentum swap, and up to `num_swaps` swaps are
+    executed. Candidate particles for swapping are those with positive
+    *x*-component momentum (flow direction) in the lower slab and those with
+    negative *x*-component momentum in the upper slab.
+
```

Comment:
Although we don't need to explain all the analysis to someone, the documentation here should include a short description about how to extract the viscosity. You haven't mentioned the `summed_exchanged_momentum` yet and what you would do with that.